addons/sale_purchase/models/product_template.py

```diff
@@ -9,23 +9,26 @@ class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
     service_to_purchase = fields.Boolean(
-        "Subcontract Service",
-        compute='_compute_service_to_purchase', store=True, readonly=False,
+        "Subcontract Service", company_dependent=True,
         help="If ticked, each time you sell this product through a SO, a RfQ is automatically created to buy the product. Tip: don't forget to set a vendor on the product.")
 
-    _sql_constraints = [
-        ('service_to_purchase', "CHECK((type != 'service' AND service_to_purchase != true) or (type = 'service'))", 'Product that is not a service can not create RFQ.'),
-    ]
-
-    @api.constrains('service_to_purchase', 'seller_ids')
+    @api.constrains('service_to_purchase', 'seller_ids', 'type')
     def _check_service_to_purchase(self):
         for template in self:
-            if template.service_to_purchase and not template.seller_ids:
-                raise ValidationError(_(
-                    "Please define the vendor from whom you would like to purchase this service automatically."))
+            if template.service_to_purchase:
+                if template.type != 'service':
+                    raise ValidationError(_("Product that is not a service can not create RFQ."))
+                elif not template.seller_ids:
+                    raise ValidationError(_("Please define the vendor from whom you would like to purchase this service automatically."))
```

Comment:
We are duplicating the text, it would be best not to.
Could you externalize it?